File: lab2-functionalprimer/python3/func_as_arg.py
def listFunc(a, b):
    # return [i for i in range(1, 6)] #Create list of ints from 1 to 5, Haskell equivalent [1..5]
    return [i for i in range(a, b)]

# def applicatorFunc(inpFunc, s):
#     if s=='s':
#         return sum(inpFunc())
#     else:
#         return sum(inpFunc())/5

def applicatorFunc(inpFunc, a, b, s):
    if s=='s':
        return sum(inpFunc(a, b))
    else:
        return sum(inpFunc(a, b))/b

print(applicatorFunc(listFunc, 1, 6, 's'))

File: lab2-functionalprimer/haskell/func_as_arg.hs
--create inpFunc
-- inpFunc = [1..5] 
inpFunc a b = [a..b]

--Define applicatorFunc
-- applicatorFunc inpFunc s = if s=='s' then sum inpFunc else (sum inpFunc)/5  
-- Guards
-- applicatorFunc :: (Ord a, Num a) => a -> Int
applicatorFunc inpFunc a b s    | s == 's' = sum (inpFunc a b)
                                | otherwise = (sum (inpFunc a b))/b

main = do
    let result = applicatorFunc inpFunc 1 20 's' --Call applicatorFunc with inpFunc and 'a' as args
    putStrLn("sum = " ++ show(result))